hack/verify-ruleguard.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script checks coding odd or unsupported patters for go language files in each
+# Kubernetes package by ruleguard.
+# Usage: `hack/verify-ruleguard.sh`.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+kube::golang::verify_go_version
+
+# Ensure that we find the binaries we build before anything else.
+export GOBIN="${KUBE_OUTPUT_BINPATH}"
+PATH="${GOBIN}:${PATH}"
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+# Install golangci-lint
+echo 'installing ruleguard '
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  go install github.com/quasilyte/go-ruleguard/cmd/ruleguard
+popd >/dev/null
+
+cd "${KUBE_ROOT}"
+
+echo 'running ruleguard '
+ruleguard -rules hack/ruleguard/rules.go -c 0 ./...
```

Comment:
does the `./...` only find uses inside k8s.io/kubernetes, or also the staging repos?

Reply:
good catch, I've double checked,  not all staging ... I need to figure it that out

Reply:
this applies to all the linters that use modules, staging folders are omitted